sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRepositoryClientLiveTests.cs

```diff
@@ -2,147 +2,180 @@
+            // Arrange
+            ContainerRepositoryClient client = CreateClient();
+            string tag = "latest";
 
-            Assert.AreEqual(_tagName, properties.Name);
+            // Act
+            TagProperties properties = await client.GetTagPropertiesAsync(tag);
+
+            // Assert
+            Assert.AreEqual(tag, properties.Name);
             Assert.AreEqual(_repositoryName, properties.Repository);
             Assert.AreEqual(new Uri(TestEnvironment.Endpoint).Host, properties.Registry);
         }
 
         [RecordedTest, NonParallelizable]
-        public async Task CanSetTagProperties([Values(true, false)] bool canList,
-                                              [Values(true, false)] bool canRead,
-                                              [Values(true, false)] bool canWrite,
-                                              [Values(true, false)] bool canDelete)
+        public async Task CanSetTagProperties()
         {
-            await _client.SetTagPropertiesAsync(
-                _tagName,
+            // Arrange
+            ContainerRepositoryClient client = CreateClient();
+            string tag = "latest";
+            TagProperties tagProperties = await client.GetTagPropertiesAsync(tag);
+            ContentProperties originalContentProperties = tagProperties.ModifiableProperties;
+
+            // Act
+            await client.SetTagPropertiesAsync(
+                tag,
                 new ContentProperties()
                 {
-                    CanList = canList,
-                    CanRead = canRead,
-                    CanWrite = canWrite,
-                    CanDelete = canDelete
+                    CanList = false,
+                    CanRead = false,
+                    CanWrite = false,
+                    CanDelete = false
                 });
 
-            TagProperties properties = await _client.GetTagPropertiesAsync(_tagName);
+            // Assert
+            TagProperties properties = await client.GetTagPropertiesAsync(tag);
+
+            Assert.IsFalse(properties.ModifiableProperties.CanList);
+            Assert.IsFalse(properties.ModifiableProperties.CanRead);
+            Assert.IsFalse(properties.ModifiableProperties.CanWrite);
+            Assert.IsFalse(properties.ModifiableProperties.CanDelete);
+
+            // Cleanup
+            await client.SetTagPropertiesAsync(tag, originalContentProperties);
+        }
+
+        [RecordedTest, NonParallelizable]
+        public async Task CanDeleteTag()
+        {
+            // Arrange
+            ContainerRepositoryClient client = CreateClient();
+            string tag = "test-delete";
+            await ImportImage(tag);
+
+            // Act
+            await client.DeleteTagAsync(tag);
+
+            // Assert
+
+            // The delete takes some time, so if we call GetTagProperties() without a delay, we get a 200 response.
+            await Task.Delay(5000);
```

Comment:
In Search and Key Vault, we have our own `DelayAsync` that effectively goes something like this:
```csharp
protected async ValueTask DelayAsync(TimeSpan? duration = null)
{
  duration ??= TimeSpan.FromSeconds(5);
  if (TestMode == RecordingTestMode.Playback)
  {
    return ValueTask.Completed;
  }
  else
  {
    await Task.Delay(duration);
  }
}
```
This way, you don't waste time during playback.

Reply:
Very cool, Heath, thanks for the pointer!

Reply:
We should probably add this to https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core.TestFramework/src/TestRecording.cs

Reply:
If so, take a look at Search or Key Vault's. It's a bit more robust (involved) than this sample, but this shows the gist of it.

Reply:
...effectively, you can optionally set a delay, or for retries (something else maybe to add, but separate) a polling interval with default. Both should probably take a `CancellationToken`, not that we use them much (but occasionally).